core/src/main/java/com/linecorp/armeria/common/HttpHeaderGetters.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.Streams;
+
+import io.netty.util.AsciiString;
+
+/**
+ * Provides the getter methods to {@link HttpHeaders} and {@link HttpHeadersBuilder}.
```

Comment:
Would users be using this interface (it's public and in common)? If so I think the javadoc could have more detail on when / how it's used.

Reply:
I'd like to hide it from the public API by making it package-local but Javadoc doesn't seem to like it. Let me try a little bit harder to hide it before giving up. :-)

Reply:
Now hidden from the public API.